substrate/client/rpc-spec-v2/src/archive/archive.rs

```diff
@@ -83,4 +112,71 @@ where
 
 		Ok(Some(hex_string(&header.encode())))
 	}
+
+	fn archive_unstable_finalized_height(&self) -> RpcResult<u32> {
+		Ok(self.client.info().finalized_number.saturated_into())
```

Comment:
I wonder if it's makes sense to throw an error or something if the block number is bigger than u32::MAX instead of truncating....

It might be impossible not sure...

Reply:
I assume it'd take a few hundred years to hit, but yeah I'd be inlined to at least panic rather than saturate since the latter is hiding some issue :)

Reply:
I've changed this to u64 to be extra safe here :D